libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/timeline/item/event/EventTimelineItemMapper.kt

```diff
@@ -78,25 +78,33 @@ fun RustEventSendState?.map(): LocalEventSendState? {
         null -> null
         RustEventSendState.NotSentYet -> LocalEventSendState.Sending
         is RustEventSendState.SendingFailed -> {
-            if (isRecoverable) {
-                LocalEventSendState.Sending
-            } else {
-                LocalEventSendState.Failed.Unknown(error)
+            when (error) {
+                QueueWedgeError.CrossVerificationRequired -> {
+                    // The current device is not cross-signed (or cross signing is not setup)
+                    LocalEventSendState.Failed.SendingFromUnverifiedDevice
+                }
+                is QueueWedgeError.IdentityViolations -> {
+                    // smart cast impossible because 'error' is a public API property declared in different module
+                    LocalEventSendState.Failed.VerifiedUserChangedIdentity((error as QueueWedgeError.IdentityViolations).users.map { UserId(it) })
```

Comment:
A way to avoid this seems to be using:

```kotlin
is RustEventSendState.SendingFailed -> error.run {
    when (this) {
        ...
        is QueueWedgeError.IdentityViolations -> {
            // smart cast impossible because 'error' is a public API property declared in different module
            LocalEventSendState.Failed.VerifiedUserChangedIdentity(this.users.map { UserId(it) })
        }
```

But I'm not sure if it's worth it.